### ISSUE TITLE
Make anonymous (PII) fields handling more efficient

### ISSUE THREAD

**Issue by npatki** [sdv-dev/SDV#439]
### Problem Description

Right now, if a user specifies that a certain column is PII we

1. Replace the column with fake data
2. Include the (fake data) column in the modeling, which means we compute a distribution & its correlations, and then
3. Override the column (again) using the provided `Faker` method

### Expected behavior

Don't model the anonymous/PII fields since they are randomly created anyways. The most efficient way to do this would be:
1. Drop the column of PII data
2. Model the _rest_ of the data (computing a distribution, correlation, etc.) and then
3. Generate Fake, PII values at the every end

Alternative; See #415 for a feature request & use case that requires mapping.

**Comment by npatki** [comment]
Marking this as stale. Our new strategy will be to implement this functionality in RDTs. Users can choose to anonymize their data fully or pseudo-anonymize it using a mapper.

See
- API doc for [AnonymizedFaker](https://docs.sdv.dev/rdt/transformers-glossary/browse-transformers/anonymizedfaker)
- proposed specs for [PseudoAnonymizedFaker](https://github.com/sdv-dev/RDT/issues/517)

**Comment by npatki** [comment]
Reopening this issue. Until the RDT changes go into effect, this is still live in the SDV.

**Comment by PJPRoche** [comment]
A further consideration would be to not include the original real data in the trained model .pkl file.  If a user only needs to supply final synthetic data, for example in the form of a .csv, then it is not a problem. But if they wish to supply a trained model .pkl file to another use so they can generate however much synthetic data they want, then it is a potential problem that the original real PII data is accessible from the .pkl

Here is an example that replicates the point

```
import cloudpickle
import pandas as pd

from sdv.tabular import CTGAN

# create dummy data
real_data = pd.DataFrame(
    data={"real_name": ["Peter", "John", "Mary", "Susan"]}
)

anonymize_fields = {
    "real_name": "name",
}

print(f"Raw data: {real_data.shape[0]} rows, {real_data.shape[1]} cols")

model = CTGAN(
    epochs=1,
    verbose=True,
    anonymize_fields=anonymize_fields,
)
model.fit(real_data)

with open("file.pkl", "wb") as output:
    cloudpickle.dump(model, output)

# delete the model to be sure it is not accessed
del model

# load back the model and inspect ANONYMIZATION_MAPPINGS
with open("file.pkl", "rb") as input:
    model_saved = cloudpickle.load(input)

print(model_saved._metadata._ANONYMIZATION_MAPPINGS)
```

which outputs:

```
Raw data: 4 rows, 1 cols
Epoch 1, Loss G:  1.5588,Loss D: -0.0033
{1636409547408: {'real_name': {'Peter': 'Jessica Reynolds', 'John': 'Jill Graham', 'Mary': 'Eric Williamson', 'Susan': 'Jordan Davis'}}}
```

containing the original names ["Peter", "John", "Mary", "Susan"]

**Comment by npatki** [comment]
Thanks @PJPRoche I can replicate what you're observing. Let's turn this into a new issue, as efficient PII handling (original topic) will be covered in future releases but saving of this info would be a separate change.

**Comment by npatki** [comment]
This issue has now been resolved in the new **[SDV 1.0 (Beta!) release]**(https://docs.sdv.dev/sdv/).

Any sensitive or PII values are now handled by an RDT, which by default drops the column before modeling -- making the overall modeling more efficient. For more info, see the [demo for custom processing and anonymization](https://colab.research.google.com/drive/1b0W9jpgOHFrkz-i1QynVuaMZXs7BSPb_?usp=sharing).
